src/MOCKernel.cpp

```diff
@@ -0,0 +1,156 @@
+#include "MOCKernel.h"
+#include "TrackGenerator.h"
+
+/**
+ * @brief Constructor for the MOCKernel
+ * @details The count for the number of segments traversed is set to zero and
+ *          the maximum optical path length is retrieved from the
+ *          TrackGenerator
+ * @param track_generator the TrackGenerator used to pull relevant tracking
+ *        data
+ */
+MOCKernel::MOCKernel(TrackGenerator* track_generator) {
+  _count = 0;
+  _max_tau = track_generator->retrieveMaxOpticalLength();
+}
+
+
+/**
+ * @brief Constructor for the VolumeKernel assigns default values, calls
+ *        the MOCKernel constructor, and pulls refernces to FSR locks and FSR
+ *        volumes from the provided TrackGenerator.
+ * @param track_generator the TrackGenerator used to pull relevant tracking
+ *        data
+ */
+VolumeKernel::VolumeKernel(TrackGenerator* track_generator) :
+                           MOCKernel(track_generator) {
+  _FSR_locks = track_generator->getFSRLocks();
+
+  if (_FSR_locks == NULL)
+    log_printf(ERROR, "Unable to create a VolumeKernel without first creating "
```

Comment:
Yes, but it's possible that you could create locks without generating Tracks. If we go to broader methods like TRRM or Multi-group Monte Carlo, then Tracks would not be generated.